### PR TITLE
Translate tutorial/stdlib2.po

### PR DESCRIPTION
**Pull request by StevenHsuYL**
This PR is continued from #61 .
The translation was completed.
Thanks for review.

Reference: https://terms.naer.edu.tw/

### CODE REVIEW

File: tutorial/stdlib2.po [hunk 1]
```diff
@@ -73,6 +93,9 @@ msgid ""
 "allows users to customize their applications without having to alter the "
 "application."
 msgstr ""
+":mod:`string` 模組包含一個多功能的 :class:`~string.Template` class，具有適合"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
":mod:`string` 模組包含一個多功能的 :class:`~string.Template` class，提供的簡化語法"
```

File: tutorial/stdlib2.po [hunk 2]
```diff
@@ -73,6 +93,9 @@ msgid ""
...
+"使用者進行編輯的簡化語法。它容許使用者客製化自己的應用程式，但不必對原應用程"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"適合使用者進行編輯時使用。它容許使用者客製化自己的應用程式，但不必對原應用程"
```

File: tutorial/stdlib2.po [hunk 3]
```diff
@@ -82,6 +105,12 @@ msgid ""
 "letters with no intervening spaces.  Writing ``$$`` creates a single escaped "
 "``$``::"
 msgstr ""
+"格式化方式是使用佔位符號名稱 (placeholder name)，它是由 ``$`` 加上合法的 "
+"Python 識別符（字母、數字和下底線）構成。若使用大括號將佔位符號包起來，就可以"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"Python 識別符（字母、數字和下底線）構成。使用大括號包覆佔位符號以允許"
```

File: tutorial/stdlib2.po [hunk 4]
```diff
@@ -82,6 +105,12 @@ msgid ""
...
+"在後面接上更多的字母和數字而無需插入空格。寫入 ``$$`` 將會跳脫為單一字元 `"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"在後面接上更多的字母和數字而無需插入空格。使用 ``$$`` 將會跳脫為單一字元 `"
```

File: tutorial/stdlib2.po [hunk 5]
```diff
@@ -91,6 +120,13 @@ msgid ""
 "meth:`~string.Template.safe_substitute` method may be more appropriate --- "
 "it will leave placeholders unchanged if data is missing::"
 msgstr ""
+"如果在 dictionary 或關鍵字引數中未提供某個佔位符號的值，那麼 :meth:`~string."
+"Template.substitute` method 將引發 :exc:`KeyError`\\ 。對於郵件合併 (mail-"
+"merge) 類型的應用程式，使用者提供的資料有可能是不完整的，此時使用 :meth:"
+"`~string.Template.safe_substitute` method 會更適當——如果資料有缺少，它會讓佔"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"`~string.Template.safe_substitute` method 會更適當——如果資料有缺少，它會保持"
```

File: tutorial/stdlib2.po [hunk 6]
```diff
@@ -91,6 +120,13 @@ msgid ""
...
+"位符號保持不變：\n"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"佔位符號不變：\n"
```

File: tutorial/stdlib2.po [hunk 7]
```diff
@@ -99,17 +135,24 @@ msgid ""
...
 "details of multiple output formats.  This makes it possible to substitute "
 "custom templates for XML files, plain text reports, and HTML web reports."
 msgstr ""
+"模板化的另一個應用，是將程式邏輯與多樣的輸出格式細節分離開來。這樣就可以為 "
+"XML 檔案、純文字報表和 HTML 網路報表替換自訂的模板。"
 
 #: ../../tutorial/stdlib2.rst:133
 msgid "Working with Binary Data Record Layouts"
-msgstr ""
+msgstr "二進制資料記錄編排 (Binary Data Record Layouts) 的處理"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
msgstr "二進制資料記錄編排 (Binary Data Record Layouts) "
```

File: tutorial/stdlib2.po [hunk 8]
```diff
@@ -120,10 +163,18 @@ msgid ""
 "\"`` represent two and four byte unsigned numbers respectively.  The ``\"<"
 "\"`` indicates that they are standard size and in little-endian byte order::"
 msgstr ""
+":mod:`struct` 模組提供了 :func:`~struct.pack` 和 :func:`~struct.unpack` 函"
+"式，用於處理不定長度的二進制記錄格式。以下範例說明，如何在不使用 :mod:"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"式，用於處理可變動長度的二進制記錄格式。以下範例說明，如何在不使用 :mod:"
```

File: tutorial/stdlib2.po [hunk 9]
```diff
@@ -133,12 +184,20 @@ msgid ""
 "background.  A related use case is running I/O in parallel with computations "
 "in another thread."
 msgstr ""
+"執行緒是一種技術，其作用是對非順序相依 (sequentially dependent) 的多個任務進"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"執行緒是一種用來對非順序相依 (sequentially dependent) 的多個任務進"
```

File: tutorial/stdlib2.po [hunk 10]
```diff
@@ -133,12 +184,20 @@ msgid ""
...
+"行解耦 (decoupling)。當其他任務在背景執行時，多執行緒可以用來增加應用程式在接"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"行解耦 (decoupling)的技術。當其他任務在背景執行時，多執行緒可以用來提升應用程式在接"
```

File: tutorial/stdlib2.po [hunk 11]
```diff
@@ -183,24 +253,35 @@ msgid ""
 "`~logging.DEBUG`, :const:`~logging.INFO`, :const:`~logging.WARNING`, :const:"
 "`~logging.ERROR`, and :const:`~logging.CRITICAL`."
 msgstr ""
+"在預設情況，資訊和除錯訊息不會被顯示，其輸出會被發送到標準錯誤 (standard "
+"error)。其他輸出選項包括，將訊息轉發到電子郵件、資料報 (datagram)、網路插座 "
+"(socket) 或 HTTP 伺服器。新的過濾器可以根據訊息的優先順序，選擇不同的選路 "
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]: Though the term was recorded in [terms.naer.edu.tw](https://terms.naer.edu.tw/detail/1285734/), '路由' may be more widely recognized nowadays.
```suggestion
"(socket) 或 HTTP 伺服器。新的過濾器可以根據訊息的優先順序，選擇不同的路由 "
```

File: tutorial/stdlib2.po [hunk 12]
```diff
@@ -183,24 +253,35 @@ msgid ""
...
+"\\ ，\\ :const:`~logging.WARNING`\\ ，\\ :const:`~logging.ERROR`\\ ，及 :"
+"const:`~logging.CRITICAL`\\ 。"
 
 #: ../../tutorial/stdlib2.rst:240
 msgid ""
 "The logging system can be configured directly from Python or can be loaded "
 "from a user editable configuration file for customized logging without "
 "altering the application."
 msgstr ""
+"日誌記錄系統可以直接從 Python 配置，也可以從使用者可編輯的配置檔案載入，以進"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"日誌記錄系統可以直接從 Python 配置，也可以透過載入使用者可編輯的配置檔案以進"
```

File: tutorial/stdlib2.po [hunk 13]
```diff
@@ -183,24 +253,35 @@ msgid ""
...
+"行客製化的日誌記錄而無須對應用程式做出變動。"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]: align the terms used at [line 98](https://github.com/python/python-docs-zh-tw/pull/80/commits/02ce49f3ae82673341c6f571d3b33f875ec4d83b#diff-7e08c9470cbb581afb7cb1e79787f3553a95a24bb9826505a43914bc54122807R98)
```suggestion
"行客製化的日誌記錄，而無須對應用程式做出變更。"
```

File: tutorial/stdlib2.po [hunk 14]
```diff
@@ -213,17 +294,27 @@ msgid ""
 "weakref objects.  Typical applications include caching objects that are "
 "expensive to create::"
 msgstr ""
+"此方式對大多數應用程式來說都沒問題，但偶爾也需要在物件仍然被其他物件所使用時"
+"持續追蹤它們。可惜的是，儘管只是追蹤它們，也會建立一個使它們永久化 "
+"(permanent) 的參照。\\ :mod:`weakref` 模組提供的工具可以不必建立參照就能追蹤"
+"物件。當該物件不再被需要時，它會自動從一個弱引用表 (weakref table) 中被移除，"
+"並為弱引用物件觸發一個回呼 (callback)。典型的應用包括暫存 (cache) 那些較佔空"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"並為弱引用物件觸發一個回呼 (callback)。典型的應用包括暫存 (cache) 那些成本較為"
```

File: tutorial/stdlib2.po [hunk 15]
```diff
@@ -213,17 +294,27 @@ msgid ""
...
+"間的物件：\n"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"昂貴的物件：\n"
```

File: tutorial/stdlib2.po [hunk 16]
```diff
@@ -256,48 +363,60 @@ msgid ""
...
+msgstr "十進制 (Decimal) 浮點數運算"
 
 #: ../../tutorial/stdlib2.rst:357
 msgid ""
 "The :mod:`decimal` module offers a :class:`~decimal.Decimal` datatype for "
 "decimal floating point arithmetic.  Compared to the built-in :class:`float` "
 "implementation of binary floating point, the class is especially helpful for"
 msgstr ""
+":mod:`decimal` 模組提供了一個 :class:`~decimal.Decimal` 資料類型，用於十進制"
+"浮點數運算。相較於內建的二進制浮點數 :class:`float` 實作，該 class 特別適用於"
```
Comment (tomatoprinx) [inline tutorial/stdlib2.po]:
```suggestion
"浮點數運算。相較於內建的二進制浮點數 :class:`float` 實作，該 class 特別適用於下列情境"
```

**Comment by StevenHsuYL** [comment]
Thanks for review! 👍

**Comment by mattwang44** [comment]
We just have a basic CI pipeline. I'll close & reopen this PR to trigger CI and see if it can pass.

**Review by mattwang44 — APPROVED** [review]
LGTM

**Review by tomatoprinx — COMMENTED** [review]
LGTM

**Review by tomatoprinx — COMMENTED** [review]
LGTM twice😂

**Pull request merged**